src/string/src/lib.rs

```diff
@@ -255,8 +255,17 @@ pub extern "C" fn strpbrk(s1: *const c_char, s2: *const c_char) -> *mut c_char {
 }
 
 #[no_mangle]
-pub extern "C" fn strrchr(s: *const c_char, c: c_int) -> *mut c_char {
-    unimplemented!();
+pub unsafe extern "C" fn strrchr(s: *const c_char, c: c_int) -> *mut c_char {
+    let len = strlen(s) as isize;
+    let c = c as i8;
+    let mut i = len - 1;
+    while *s.offset(i) != 0 && i >= 0 {
```

Comment:
Just a random question: What is the purpose of checking for the null character?

Reply:
Good point. I think it’s uneceasary.

Reply:
:+1: thanks for working on this!